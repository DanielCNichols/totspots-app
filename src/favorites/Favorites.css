.favorite h3 {
  text-align: center;
}

.favorite button {
  display: block; 
  margin: 30px auto;
  border-radius: 17px;
  border-color: #ED254E;
  background-color: transparent;
  width: 140px;
  height: 40px;
  font-size: 1.1em;
}


@media screen and (min-width: 800px) {
  .favorite button {
    width: 100px;
    font-size: 1em;
  }

  .favorite button:hover {
    color: white;
    background-color: #ED254E;
    cursor: pointer;
  }
}