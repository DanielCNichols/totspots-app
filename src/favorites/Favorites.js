import React from 'react'
import ApiService from '../services/api-service';
import VenuesContext from '../VenuesContext'
import './Favorites.css'
import {Link} from 'react-router-dom'

class Favorites extends React.Component {
  static contextType = VenuesContext

  deleteFavorite(id, callback) {
    ApiService.deleteFavorite(id)
    .then((favorites) => {
      callback(id);
    })
    .catch((this.context.setError))
  }

  render() {
    let { favorites } = this.props;
    return (
      <li className="favorite" key={favorites.id}>
        <h3>{favorites.venue_name}</h3>
        <p>{favorites.venue_type}</p>
        <p>{favorites.address}</p>
        <span>{favorites.city}</span>, <span>{favorites.state} </span> 
        <span>{favorites.zipcode}</span>
        <p>{favorites.url}</p>
        <p>{favorites.phone}</p>
        <button onClick={() => this.deleteFavorite(favorites.id, this.context.deleteFavorite)}>Remove</button>
      </li>
    );
  }
}

export default Favorites;
