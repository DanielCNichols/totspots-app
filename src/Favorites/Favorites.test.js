import React from 'react';
import ReactDOM from 'react-dom';
import Favorites from './Favorites.js';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';


let favorite= { 
    venue_name: "Early Bird",
    city: "Durham",
    state: "NC",
    address: "987 Erwin Rd.",
    zipcode: "27705",
    venue_type: "Restaurant",
    id: 5,
    url: null,
    phone: "9848880417"
}

it('renders Favorites.js without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MemoryRouter>
    <Favorites favorites={favorite}/>, div);
  </MemoryRouter>, div)
  ReactDOM.unmountComponentAtNode(div);
});


it('renders Favorites as expected', () => {
  const tree = renderer
  .create(
    <MemoryRouter>
      <Favorites name="Favorites" favorites={favorite}/>
    </MemoryRouter>)
  .toJSON();
  expect(tree).toMatchSnapshot();
})
