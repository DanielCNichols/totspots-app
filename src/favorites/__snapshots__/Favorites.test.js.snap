// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Favorites as expected 1`] = `
<li
  className="favorite"
>
  <h3>
    Early Bird
  </h3>
  <div
    className="favorite_info"
  >
    <p
      className="type"
    >
      Type: 
      Restaurant
    </p>
    <p>
      Address: 
      987 Erwin Rd.
      <span
        className="city"
      >
        Durham
      </span>
      ,
       
      <span
        className="state"
      >
        NC
         
      </span>
      <span>
        27705
      </span>
    </p>
    <p
      className="link"
    >
      Website:
      <a
        href={null}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
            
        </span>
      </a>
    </p>
    <p>
      Phone: 
      9848880417
    </p>
  </div>
  <div
    className="favorites_controls"
  >
    <button
      className="visit"
      onClick={[Function]}
    >
      See more
    </button>
    <button
      className="delete_favorite"
      onClick={[Function]}
    >
      Remove
    </button>
  </div>
</li>
`;
